community/en/tiny_imagenet_custom_resnet.ipynb

```diff
@@ -0,0 +1,552 @@
+    "\n",
+    "        x = BatchNormalization(axis=chanDim, epsilon=bnEps, momentum=bnMom)(x)\n",
+    "        x = Activation(\"relu\")(x)\n",
+    "        x = AveragePooling2D((8, 8))(x)\n",
+    "  \n",
+    "        x = Conv2D(200, (1,1), kernel_regularizer=l2(reg))(x)\n",
+    "        x = Flatten()(x)\n",
+    "        x = Activation(\"softmax\")(x)\n",
+    "\n",
+    "        model = tf.keras.Model(inputs, x, name=\"resnet\")\n",
+    "\n",
+    "        return model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = ResNet.build(64, 64, 3, 200, (3, 4, 6), (64, 128, 256, 512), reg=0.0005)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model.summary()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Compile the model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from tensorflow.keras.optimizers import Adam\n",
+    "model.compile(loss=\"categorical_crossentropy\", optimizer=opt, metrics=[\"accuracy\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Using fit_generator to train the model\n",
+    "ImageDataGenerator is best suited for augmenting images on the fly and training the model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model.fit_generator(\n",
+    "  train_generator,\n",
+    "  steps_per_epoch=100000 // 64,\n",
+    "  validation_data=val_generator,\n",
+    "  validation_steps=10000 // 64,\n",
+    "  epochs=5,\n",
+    "  max_queue_size=64 * 2,\n",
+    "  callbacks=callbacks, verbose=1\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### List of references for easy lookup\n",
+    "\n",
+    "---\n",
+    "\n",
+    "1. Building blocks of interpretability: [Link](https://distill.pub/2018/building-blocks/) (Holy Grail of Intuition!)\n",
+    "2. Deep Residual Learning for image classification: [Link](https://arxiv.org/abs/1512.03385) (Resnet Paper)\n",
+    "3. Bag of tricks for image classification: [Link](https://arxiv.org/abs/1812.01187) (Tweaks and tricks to Resnet for increased performance paper)\n",
+    "2. Imbalanced Deep Learning by Minority Class\n",
+    "Incremental Rectification: [Link](https://arxiv.org/pdf/1804.10851.pdf) (Selectively Sampling Data paper)\n",
+    "2. Improved Regularization of Convolutional Neural Networks with Cutout: [Link](https://arxiv.org/pdf/1708.04552.pdf) (Cutout/Occlusion Augmentation paper)\n",
+    "3. Survey of resampling techniques for improving\n",
+    "classification performance in unbalanced datasets [Link](https://arxiv.org/pdf/1608.06048v1.pdf) (Resampling paper)"
+   ]
```

Comment:
Delete everything in this file below here.

Reply:
Done. This will reflect in the next commit.